pkg/manifest/ostree_deployment.go

```diff
@@ -344,7 +344,8 @@ func (p *OSTreeDeployment) serialize() osbuild.Pipeline {
 		}
 	}
 
-	if !hasRoot {
+	if !hasRoot && p.ostreeSpec != nil {
+		// NOTE: fails on container-based deployments
```

Comment:
Do we want to error out here instead if it fails anyhow?

Reply:
Moved it to the image function so it works like the other change we made with the kernel options.